pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -852,6 +856,15 @@ func (gce *GCECloud) createTargetPool(name, serviceName, region string, hosts []
 	for _, host := range hosts {
 		instances = append(instances, makeHostURL(gce.projectID, host.Zone, host.Name))
 	}
+	if len(instances) > maxInstancesPerTargetPool {
```

Comment:
You can check for node health like https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L3048